python/lsst/pipe/tasks/multiBand.py

```diff
@@ -819,6 +898,83 @@ def __init__(self, butler=None, schema=None, peakSchema=None, refObjLoader=None,
+        if config.doMatchSources is False:
+            outputDatasetTypes.pop("matchResult")
+        if config.doWriteMatchesDenormalized is False:
+            outputDatasetTypes.pop("denormMatches")
+        return outputDatasetTypes
+
+    def getInitOutputDatasets(self):
+        return {"outputSchema": afwTable.SourceCatalog(self.schema)}
+
+    def adaptArgsAndRun(self, inputData, inputDataIds, outputDataIds, butler):
+        refObjLoader = ReferenceObjectLoader(inputDataIds['refCat'], butler,
+                                             config=self.config.refObjLoader, log=self.log)
+        self.match.setRefObjLoader(refObjLoader)
+
+        # Set psfcache
+        inputData['exposure'].getPsf().setCacheCapacity(self.config.psfCache)
+
+        # Transform inputCatalog
+        idFactory = afwTable.IdFactory.makeSimple()
+        table = afwTable.SourceTable.make(self.schema, idFactory)
+        sources = afwTable.SourceCatalog(table)
+        sources.extend(inputData.pop('intakeCatalog'), self.schemaMapper)
+        table = sources.getTable()
+        table.setMetadata(self.algMetadata)  # Capture algorithm metadata to write out to the source catalog.
+        inputData['sources'] = sources
+
+        inputData['exposureId'] = 0
+
+        skyMap = inputData.pop('skyMap')
+        tractNumber = inputDataIds['intakeCatalog']['tract']
+        tractInfo = skyMap[tractNumber]
+        patchInfo = tractInfo.getPatchInfo(inputDataIds['intakeCatalog']['patch'])
+        skyInfo = Struct(
+            skyMap=skyMap,
+            tractInfo=tractInfo,
+            patchInfo=patchInfo,
+            wcs=tractInfo.getWcs(),
+            bbox=patchInfo.getOuterBBox()
+        )
+        inputData['skyInfo'] = skyInfo
+
+        if self.config.doPropagateFlags:
+            # Filter out any visit catalog that is not coadd inputs
+            ccdInputs = inputData['exposure'].getInfo().getCoaddInputs().ccds
+            visitKey = ccdInputs.schema.find("visit").key
+            ccdKey = ccdInputs.schema.find("ccd").key
+            inputVisitIds = set()
+            ccdRecordsWcs = {}
+            for ccdRecord in ccdInputs:
+                visit = ccdRecord.get(visitKey)
+                ccd = ccdRecord.get(ccdKey)
+                inputVisitIds.add((visit, ccd))
+                ccdRecordsWcs[(visit, ccd)] = ccdRecord.getWcs()
+
+            inputCatalogsToKeep = []
+            inputCatalogWcsUpdate = []
+            for (i, dataId) in enumerate(inputDataIds['visitCatalogs']):
+                key = (dataId['visit'], dataId['detector'])
+                if (key) in inputVisitIds:
+                    inputCatalogsToKeep.append(inputData['visitCatalogs'][i])
+                    inputCatalogWcsUpdate.append(ccdRecordsWcs[key])
+            inputData['visitCatalogs'] = inputCatalogsToKeep
+            inputData['wcsUpdates'] = inputCatalogWcsUpdate
+            inputData['ccdInputs'] = ccdInputs
```

Comment:
Hopefully we can clean this up more fully when we don't have to support the `CmdLineTask` versions of things anymore, but do you think it would be an improvement now to bundle these three arguments into (e.g.) a `namedtuple` or something.

Reply:
Since ccdInputs is in the interface for the existing propagate visit flags, this is something I dont want to bundle as it would just need to be unpacked later. That leaves 2 things which does not seem worth bundling. I think it is better to have an explicit function signature instead of packing things just to unpack them inside the function.